lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -194,6 +194,40 @@ func (d *zfs) getDatasets(dataset string, types string) ([]string, error) {
 	return children, nil
 }
 
+func (d *zfs) filterRedundantOptions(dataset string, options ...string) ([]string, error) {
+	keys := []string{}
+	values := []string{}
+
+	// Extract keys and values from options.
+	for _, option := range options {
+		property := strings.Split(option, "=")
+
+		if len(property) != 2 {
+			return nil, fmt.Errorf("Wrongly formatted option \"%s\"", option)
```

Comment:
We normally use `%q` as it will take care of quoting the value:

```suggestion
			return nil, fmt.Errorf("Wrongly formatted option %q, option)
```